File: modelo/docenteexamen.cs
using System.Data;
using System;
    public class docenteexamen : oObjeto
    {
        public int Iddocente{ get; set; }
        public int Idexamen { get; set; }
        public String Observacion{set;get;}
        public docenteexamen(){}
        public docenteexamen(DataRow dr)
    {
        this.Id = Convert.ToInt32(dr["id"]);
        this.Iddocente= Convert.ToInt32(dr["iddocente"]);
        this.Idexamen= Convert.ToInt32(dr["idexamen"]);
        this.Observacion= dr["observacion"].ToString();
    }

    }
 

File: ClientApp/src/app/clases/docenteexamen.ts
export class docenteexamen{
    id: number;
    iddocente : number;
    idexamen : number;
    observacion: string;
    
    
    constructor(obj:any){
        //el orden de os parametros afecta la vista de los input en el modal
        this.id = +obj.id;
        this.iddocente= +obj.iddocente;
        this.idexamen= +obj.idexamen;
        this.observacion=obj.observacion;
         // formateo a number el string // parseInt(val

    }
    
}